utils/staking-miner/src/monitor.rs

```diff
@@ -54,17 +63,63 @@ async fn ensure_no_previous_solution<
 	T: EPM::Config + frame_system::Config<AccountId = AccountId>,
 	B: BlockT,
 >(
-	ext: &mut Ext,
+	client: &WsClient,
+	at: B::Hash,
 	us: &AccountId,
 ) -> Result<(), Error<T>> {
-	use EPM::signed::SignedSubmissions;
-	ext.execute_with(|| {
-		if <SignedSubmissions<T>>::get().iter().any(|ss| &ss.who == us) {
-			Err(Error::AlreadySubmitted)
-		} else {
-			Ok(())
+	let indices_key = storage_value(EPM_MODULE_PREFIX, b"SignedSubmissionIndices");
```

Comment:
hard-coding `EPM_MODULE_PREFIX` is wrong, since it is coming from the name of the pallet in `construct_runtime`, and it could be different. 

You can however get this from `T`: https://paritytech.github.io/substrate/master/frame_support/traits/trait.PalletInfo.html#tymethod.name